crates/rpc/rpc/src/trace.rs

```diff
@@ -308,11 +310,21 @@ where
         }
 
         let block_traces = futures::future::try_join_all(block_traces).await?;
-        let all_traces = block_traces
+        let mut all_traces = block_traces
             .into_iter()
             .flatten()
             .flat_map(|traces| traces.into_iter().flatten().flat_map(|traces| traces.into_iter()))
-            .collect();
+            .collect::<Vec<_>>();
+
+        // include all the block reward traces
+        let reward_blocks =
+            blocks.iter().map(|block| self.inner.eth_api.block_by_id(block.number.into()));
+
+        for block in (futures::future::try_join_all(reward_blocks).await?).into_iter().flatten() {
```

Comment:
none of this will be required with my suggestions, we already have the block

we could also further optimize this and stop looping over blocks if we find one that is past the merge